browser/brave_shields/ad_block_subscription_download_manager_getter.cc

```diff
@@ -70,37 +68,43 @@ class AdBlockSubscriptionDownloadManagerFactory
 };
 
 class AdBlockSubscriptionDownloadManagerGetterImpl
-    : public ProfileManagerObserver {
+    : public ProfileManagerObserver,
+      public base::SupportsWeakPtr<
+          AdBlockSubscriptionDownloadManagerGetterImpl> {
  public:
   AdBlockSubscriptionDownloadManagerGetterImpl(
       base::OnceCallback<void(AdBlockSubscriptionDownloadManager*)> callback)
       : callback_(std::move(callback)) {
+    g_browser_process->profile_manager()->AddObserver(this);
+    // trigger a check for the download manager
+    OnProfileAdded(nullptr);
```

Comment:
I suggest we just check whether the profile is ready in `AdBlockSubscriptionDownloadManagerGetter()` and directly call the callback in that function if it is ready.

If it is not - we create this observer that does `AddObserver()` and waits for either `OnProfileAdded()` or `OnProfileManagerDestroying()`. This way we can avoid posting tasks, using weak pointers and other unneeded complexity that is absolutely redundant for this trivial task.

Reply:
if we don't PostTask, how can we delete the class?

Reply:
https://github.com/brave/brave-core/pull/10203